--- src/codegen_llvm/translate/flow.c
+++ src/codegen_llvm/translate/flow.c
@@ -156,7 +156,7 @@
             LLVMTypeRef ret_ty = LLVMGetReturnType(LLVMGlobalGetValueType(current_func));
             
             if (op1) {
-                if (LLVMGetTypeKind(ret_ty) == LLVMStructTypeKind && LLVMGetTypeKind(LLVMTypeOf(op1)) == LLVMPointerTypeKind) {
+                if ((LLVMGetTypeKind(ret_ty) == LLVMStructTypeKind || LLVMGetTypeKind(ret_ty) == LLVMArrayTypeKind) && LLVMGetTypeKind(LLVMTypeOf(op1)) == LLVMPointerTypeKind) {
-                    if (inst->op1 && inst->op1->type.base == TYPE_CLASS) {
+                    if (inst->op1 && (inst->op1->type.base == TYPE_CLASS || inst->op1->type.array_size > 0)) {
                         VarType base_t = inst->op1->type;
                         base_t.ptr_depth = 0;
                         LLVMTypeRef struct_ty = get_llvm_type(ctx, base_t);
                         if (struct_ty) {
                             op1 = LLVMBuildLoad2(ctx->builder, struct_ty, op1, "load_ret");
                         }
                     }
                 }
