--- src/codegen_llvm/translate/stmt.c
+++ src/codegen_llvm/translate/stmt.c
@@ -32,7 +32,7 @@
                         size = LLVMBuildZExt(ctx->builder, size, LLVMInt64TypeInContext(ctx->llvm_ctx), "sz_ext");
                     }
-                    // res = LLVMBuildAlloca(ctx->builder, LLVMInt8TypeInContext(ctx->llvm_ctx), "alloc");
-                    res = LLVMBuildArrayAlloca(ctx->builder, LLVMInt8TypeInContext(ctx->llvm_ctx), size, "alloc");
+                    LLVMValueRef const_size = LLVMConstInt(LLVMInt64TypeInContext(ctx->llvm_ctx), 10, 0);
+                    res = LLVMBuildArrayAlloca(ctx->builder, LLVMInt8TypeInContext(ctx->llvm_ctx), const_size, "alloc");
                 } else {
                     res = LLVMBuildAlloca(ctx->builder, ty, "alloc");
                 }
