--- src/codegen_llvm/translate/stmt.c
+++ src/codegen_llvm/translate/stmt.c
@@ -31,7 +31,8 @@
                     if (LLVMGetTypeKind(LLVMTypeOf(size)) == LLVMIntegerTypeKind && LLVMGetIntTypeWidth(LLVMTypeOf(size)) < 64) {
                         size = LLVMBuildZExt(ctx->builder, size, LLVMInt64TypeInContext(ctx->llvm_ctx), "sz_ext");
                     }
-                    res = LLVMBuildArrayAlloca(ctx->builder, LLVMInt8TypeInContext(ctx->llvm_ctx), size, "alloc");
+                    // res = LLVMBuildArrayAlloca(ctx->builder, LLVMInt8TypeInContext(ctx->llvm_ctx), size, "alloc");
+                    res = LLVMBuildAlloca(ctx->builder, LLVMInt8TypeInContext(ctx->llvm_ctx), "alloc");
                 } else {
                     res = LLVMBuildAlloca(ctx->builder, ty, "alloc");
                 }
