--- src/codegen_llvm/translate/stmt.c
+++ src/codegen_llvm/translate/stmt.c
@@ -293,5 +293,9 @@
                             } else if (LLVMIsOpaqueStruct(base_ty)) {
                                 fprintf(stderr, "GEP into unsized type!\n");
                             }
+                            unsigned num_elems = LLVMCountStructElementTypes(base_ty);
+                            if ((unsigned)inst->op2->val.int_val >= num_elems) {
+                                fprintf(stderr, "GEP index out of bounds! %s has %u fields, requested %u\n", ptr_t.class_name, num_elems, (unsigned)inst->op2->val.int_val);
+                            }
                             res = LLVMBuildStructGEP2(ctx->builder, base_ty, op1, (unsigned)inst->op2->val.int_val, "struct_gep");
                         }
